File: src/main/java/Library/Accounts.java
package Library;

import java.util.ArrayList;


//  An array to store the borrower accounts
public class Accounts {

    // A list when initializing
    private ArrayList<Borrower> list = new ArrayList<>();

    // Add the borrower when initializing
    void add(Borrower b){
        list.add(b);
    }

    // return the size of list
    public int size(){
        return list.size();
    }

    // return all borrowers
    public ArrayList<Borrower> all(){
        return list;
    }

    // Find the specific borrower username
    public Borrower getUser(String username) {
        for (Borrower b : list) {
            if (b.getUsername().equals(username)) {
                return b;
            }
        }
        return null;
    }

}

File: A3/models/Accounts.js
// Accounts.js

class Accounts {
    constructor() {
        this.list = []; // Array of Borrower objects
    }

    // add(Borrower b)
    add(borrower) {
        this.list.push(borrower);
    }

    // size()
    size() {
        return this.list.length;
    }

    // all() -> returns original list (NOT a copy)
    all() {
        return this.list;
    }

    // getUser(username)
    getUser(username) {
        for (const b of this.list) {
            if (b.getUsername() === username) {
                return b;
            }
        }
        return null;
    }
}

module.exports = Accounts;
